pkg/doc/verifiable/credential.go

```diff
@@ -464,10 +478,12 @@ func WithJSONLDValidation() CredentialOpt {
 }
 
 // WithBaseContextValidation validates that only the fields and values (when applicable) are present
-// in the document. No extra fields are allowed (outside of credentialSubject).
+// in the document. No extra fields are allowed (outside of credentialSubject). Strict JSON Schema validation is made
+// (i.e. fields not defined in the base JSON Schema are not allowed).
 func WithBaseContextValidation() CredentialOpt {
 	return func(opts *credentialOpts) {
 		opts.modelValidationMode = baseContextValidation
+		opts.strictValidation = true
```

Comment:
i am not sure that this line works without flexibility in the credentialsubject.

Reply:
Corrected.